src/domain/profile/ProfileProvider.tsx

```diff
@@ -1,89 +1,66 @@
 import React from 'react';
-import { useSelector } from 'react-redux';
-import { useOidcClient } from 'hds-react';
 
+import { ProfileContext, ProfileType } from './ProfileContext';
+import useProfileFetcher from './hooks/useProfileFetcher';
 import LoadingSpinner from '../../common/components/spinner/LoadingSpinner';
-import { ProfileContext } from './ProfileContext';
-import useProfile from './hooks/useProfile';
-import { MyProfile } from './types/ProfileQueryTypes';
-import { useIsFullyLoggedIn } from '../auth/useIsFullyLoggedIn';
-import { profileSelector } from './state/ProfileSelectors';
-import { clearProfile as clearProfileFromRedux } from './state/ProfileActions';
-import { persistor } from '../app/state/AppStore';
 
 export default function ProfileProvider({
   children,
 }: {
   children: React.ReactNode;
 }) {
-  // TODO: Get rid of profile redux storage
-  // Must be used here now in order to get the profile for component's first render.
-  const reduxStorageProfile = useSelector(profileSelector);
-  const [profile, setProfileToContext] = React.useState<MyProfile | null>(
-    reduxStorageProfile?.id ? reduxStorageProfile : null
-  );
-  const [loadingProfileToContext, setLoadingProfileToContext] =
-    React.useState(true);
-
-  const { isAuthenticated } = useOidcClient();
-  const [isLoginReady, loginInProgress] = useIsFullyLoggedIn();
-
-  const [fetchProfile, { data, loading, refetch, called }] = useProfile({
-    setProfileToContext: setProfileToContext,
-    setLoading: setLoadingProfileToContext,
+  const [profile, setProfileToContext] = React.useState<ProfileType>(null);
+  const {
+    refetch,
+    loading: isProfileFetcherLoading,
+    called: isFetchCalled,
+  } = useProfileFetcher({
+    setProfileToContext,
   });
 
-  const isLoading =
-    loginInProgress || loading || (loadingProfileToContext && called);
-
-  const clearProfile = () => setProfileToContext(null);
-
   const refetchProfile = React.useCallback(() => {
-    if (refetch && data?.myProfile)
-      refetch().then(({ data: refreshedData }) => {
-        setProfileToContext(refreshedData?.myProfile || null);
-      });
-  }, [data?.myProfile, refetch]);
+    refetch().then(({ data: refreshedData }) => {
+      setProfileToContext(refreshedData?.myProfile || null);
+    });
+  }, [refetch]);
+
+  const clearProfile = React.useCallback(() => setProfileToContext(null), []);
 
   const contextValue = React.useMemo(
     () => ({
       profile,
       clearProfile,
       updateProfile: setProfileToContext,
       refetchProfile,
-      loading: isLoading,
-      fetchCalled: called,
+      isLoading: isProfileFetcherLoading,
+      isFetchCalled,
     }),
-    [called, isLoading, profile, refetchProfile]
+    // The profile object is always changing when the isFullyLoggedIn runs
```

Comment:
As the useIsFullyLoggedIn hook was deleted so might this comment be out-of-sync?